### PR TITLE
New TOC updates for RC2

### PR DESCRIPTION
**Pull request by thesmithee**
Add new TOC updates, mostly around the .NET Core general topics. Also, change
the TOC structure to resemble the final thing we want in RC2/RTM. With this change,
the TOC is split into these major groups:
- .NET Core concepts -> .NET Core general topics like StdLib, types of apps etc.
- Scenarios -> various application scenarios; console and lib writing were moved here
- Learn .NET -> variouis topics on general .NET development
- Learn C# -> self-explanatory

Some folders and files were moved because of this change, so please pay attention.

/cc @richlander @cartermp @mairaw @stevehoag @rpetrusha

### CODE REVIEW

**Comment by stevehoag** [comment]
LGTM

**Comment by cartermp** [comment]
LGTM

File: docs/toc.md [hunk 1]
```diff
@@ -24,40 +50,19 @@
 ## [🔧 Reflection & code generation](languages/csharp/reflection.md)
```
Comment (mairaw) [inline docs/toc.md]: Line 47 - should we remove the wrench from that one? Looks like asynchronous programming already has content

File: docs/toc.md [hunk 2]
```diff
@@ -68,20 +73,18 @@
...
+### [How to: Add and Remove Items from a ConcurrentDictionary](essentials/collections/threadsafe/howtoAddandRemoveItems.md)
 ### [How to: Add and Take Items Individually from a BlockingCollection](essentials/collections/threadsafe/howtoAddandTakeItems.md)
 ### [How to: Add Bounding and Blocking Functionality to a Collection](essentials/collections/threadsafe/howtoAddBoundingandBlocking.md)
 ### [How to: Use ForEach to Remove Items in a BlockingCollection](essentials/collections/threadsafe/howtoUseForEachtoRemove.md)
 ### [How to: Use Arrays of Blocking Collections in a Pipeline](essentials/collections/threadsafe/howtoUseArraysofBlockingCollections.md)
 ### [How to: Create an Object Pool by Using a ConcurrentBag](essentials/collections/threadsafe/howtoCreateanObjectPool.md)
+### [🔧 Common Language Runtime (CLR)](essentials/clr.md)
+### [🔧 Class Library overview](essentials/classlibrary.md)
+### [🔧 Working with base types](essentials/basetypes.md)
+### [🔧 Exception management in .NET Core](essentials/exceptions.md)
```
Comment (mairaw) [inline docs/toc.md]: This is also done, no? Remove the wrench? It should have the sub-topics like we have for the collections.

**Comment by mairaw** [comment]
LGTM modulo the feedback

File: docs/apis/index.md [hunk 3]
```diff
@@ -1,14 +1,14 @@
 
-# Application scenarios
+# Working with APIs
```
Comment (richlander) [inline docs/apis/index.md]: I'm wondering if "Using APIs" is a better title. It should be "work" ;)

File: docs/toc.md [hunk 4]
```diff
@@ -10,6 +9,33 @@
 ### [🔧 Development using vim](getting-started/devenv/using-vim.md)
 ### [🔧 Development using Sublime Text](getting-started/devenv/using-sublime.md)
 ## [Writing your first console application on .NET Core](getting-started/first-console-app.md)
+# [Scenarios](scenarios/index.md)
+## [Writing console applications](scenarios/console/index.md)
+### [🔧 Overview and moving pieces](scenarios/console/overview.md)
+### [🔧 Console applications paradigm](scenarios/console/paradigm.md)
```
Comment (richlander) [inline docs/toc.md]: I was thinking that there would be very limited conceptual content under "scenarios". All the conceptual content should be under "Concepts". I was thinking that "Scenarios" would exclusively be "Here is how you do topical scenario X + some best practice guidance" and of course link to the concepts section as needed.

For scenarios, I am hoping we eventually grow into more narrow scenarios, like  "Making a Windows Service".

**Comment by richlander** [comment]
LGTM modulo the feedback

File: docs/toc.md (hunk 4 above)
Comment (blackdwarf) [inline docs/toc.md]: @richlander Kewl, makes sense. This is just moved what we currently have. I can re-arrange any way we see/feel fit.

File: docs/apis/index.md (hunk 3 above)
Comment (blackdwarf) [inline docs/apis/index.md]: Fixed.

File: docs/toc.md (hunk 1 above)
Comment (blackdwarf) [inline docs/toc.md]: Fixed.

**Comment by blackdwarf** [comment]
New version is up @mairaw @richlander please take a look so I can merge posthaste! :)

**Comment by mairaw** [comment]
I still don't see the exception section expanded or am I missing something?

**Comment by blackdwarf** [comment]
@mairaw it's not expanded. :) I don't think that thing got merged yet?

**Comment by mairaw** [comment]
I do see the topics at the master branch: https://github.com/dotnet/core-docs/tree/master/docs/essentials/exceptions

**Comment by blackdwarf** [comment]
Hm, maybe the author of said docs should do it?

**Comment by mairaw** [comment]
@kempb, was it you who worked on the exceptions topics?
So merge like this and then we can add the exception TOC later on.

**Comment by richlander** [comment]
LGTM

**Comment by blackdwarf** [comment]
Merging. All other will have to rebase and fix. :)

**Comment by cartermp** [comment]
'Twas @kempb who worked on it.  Let's get that added later on.  The exceptions topic needs a but more work before it's capable of being merged in

**Comment by cartermp** [comment]
and as I type this @blackdwarf handles it like a boss 😄

**Comment by blackdwarf** [comment]
Like a baws... :D

**Pull request merged**